src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -518,3 +524,169 @@ export const concatAttachmentsAndResponses = (
+      : answer
+  })
+  return {
+    question: field.question,
+    answerTemplate: maskedAnswerTemplate,
+  }
+}
+
+/**
+ * Function to extract information for email json field from response
+ */
+const getJsonFormattedResponse = (
+  response: ResponseFormattedForEmail,
+): EmailJsonField | undefined => {
+  const { answer, fieldType } = response
+  // Headers are excluded from JSON data
+  if (fieldType !== BasicField.Section) {
+    const jsonData = {
+      question: getJsonPrefixedQuestion(response),
+      answer,
+    }
+    return jsonData
+  }
+  return undefined
+}
+
+/**
+ * Function to extract information for email form field from response
+ */
+const getFormFormattedResponse = (
+  response: ResponseFormattedForEmail,
+  hashedFields: Set<string>,
+): EmailFormField => {
+  const { answer, fieldType } = response
+  const answerSplitByNewLine = answer.split('\n')
+  const formData = {
+    question: getFormDataPrefixedQuestion(response, hashedFields),
+    answerTemplate: answerSplitByNewLine,
+    answer,
+    fieldType,
+  }
+  return formData
+}
+
+/**
+ * Function to extract information for email autoreply field from response
+ */
+const getAutoReplyFormattedResponse = (
+  response: ResponseFormattedForEmail,
+): EmailAutoReplyField | undefined => {
+  const { question, answer, isVisible } = response
+  const answerSplitByNewLine = answer.split('\n')
+  // Auto reply email will contain only visible fields
+  if (isVisible) {
+    const autoReplyData = {
+      question, // No prefixes for autoreply
+      answerTemplate: answerSplitByNewLine,
+    }
+    return autoReplyData
+  }
+  return undefined
+}
+
+export class EmailDataObj {
+  parsedResponses: ProcessedFieldResponse[]
+  hashedFields: Set<string>
+
+  constructor(
+    parsedResponses: ProcessedFieldResponse[],
+    hashedFields: Set<string>,
+  ) {
+    this.parsedResponses = parsedResponses
+    this.hashedFields = hashedFields
+  }
+
+  /**
+   * Getter function to return jsonData
+   */
+  get jsonData(): EmailJsonField[] {
+    return this.parsedResponses.flatMap((response) =>
+      createFormattedDataForOneField(
+        response,
+        this.hashedFields,
+        getJsonFormattedResponse,
+      ).filter(notUndefined),
+    )
+  }
+
+  /**
+   * Getter function to return autoReplyData
+   */
+  get autoReplyData(): EmailAutoReplyField[] {
+    return this.parsedResponses.flatMap((response) =>
+      createFormattedDataForOneField(
+        response,
+        this.hashedFields,
+        getAutoReplyFormattedResponse,
+      ).filter(notUndefined),
+    )
+  }
+
+  /**
+   * Getter function to return maskedAutoReplyData
+   * Returns a masked version of autoReplyData
+   */
+  get maskedAutoReplyData(): EmailAutoReplyField[] {
+    return this.autoReplyData.map((autoReplyField) => {
+      return autoReplyField.question === SPCPFieldTitle.CpUid
+        ? maskField(autoReplyField)
+        : autoReplyField
```

Comment:
what happens if a user creates a form field with precisely the same question title as `SPCPFieldTitle.CpUid`? will the code inadvertently mask the data?

Reply:
Yes, but only if this happens on a Corppass form, where there would be 2 fields with the exact same name - I can't see why any user would want to do this

Reply:
Be that as it may, as a matter of principle we should not be shipping software that relies on the users' cooperation to function correctly.